src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -216,9 +217,9 @@ public ValueCollection Values
             get
             {
                 int i = FindEntry(key);
-                if (i >= 0) return _entries[i].value;
+                if (i >= 0) return _entries![i].value;
                 ThrowHelper.ThrowKeyNotFoundException(key);
-                return default;
+                return default!; // TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/26761 (annotating ThrowHelper removes this return statement).
```

Comment:
Ditto, https://github.com/dotnet/csharplang/issues/538

Reply:
Alternatively the code could be changed to:
```C#
int i = FindEntry(key);
if (i < 0) ThrowHelper.ThrowKeyNotFoundException(key);
return _entries![i].value;
```
but I think what you've got is better, just in case the inversion led to an unexpected regression.

Reply:
I rather not change code to avoid an unexpected regression, but it looks like i < 0 will only be true when the Entry is not found.